### ISSUE TITLE
Hard Stop if `id` is missing from pool result?

### ISSUE THREAD

**Issue by brandonlehmann** [turtlecoin/violetminer#13]
Appears some pools don't return responses correctly:

```
[cryptonote.social:4444] Error: Failed to parse message from pool ([json.exception.out_of_range.403] key 'id' not found) ({"jsonrpc":"2.0","method":"job","params":{"blob":"0000babbaaeb05000000000000000000000000000000000000000000000000000000000000000000000000106d1a571c48e8f70d52e7074311c40ead574233e35f3008e48451532d3a49e601","job_id":"551503","target":"3d2f0200","algo":"chukwa","blockMajorVersion":6,"blockMinerVersion":0}}
)```

**Comment by zpalmtree** [comment]
I cannot reproduce this. I am successfully connected to cryptonote social and submitting shares. Appears to be a bug with the `share accepted` message being sent multiple times, however.

**Comment by brandonlehmann** [comment]
Use a different wallet with the same username and it throws a fit :) (their implementation sure, but I did see a proper log message and then this)

**Comment by zpalmtree** [comment]
I tried that, and it gave me the log message, but not the error you are getting.
![image](https://user-images.githubusercontent.com/22151537/64066726-46759000-cc15-11e9-8daa-0bf188cb0455.png)

**Comment by zpalmtree** [comment]
I can just make the id property optional which will fix this either way. I don't currently use it for anything.

**Comment by zpalmtree** [comment]
Resolved by fe16732c1e15bcad9d9f97e6bfcfd8cc2c888a01
